genai-perf/genai_perf/parser.py

```diff
@@ -896,6 +897,8 @@ def profile_handler(args, extra_args) -> None:
         telemetry_data_collector=telemetry_data_collector,
     )
 
+    return telemetry_data_collector
```

Comment:
This doesnt feel like the right place for this.
I think we need to decouple the telemetry collector creation from the profile command.

Reply:
The reason to create telemetry_data_collector in profile command is because we need to collect while PA is running.
And the return is because telemetry_data_collector had metrics decoupled, and so it calls the telemetry_statistics on the metrics.
Something that I could think of it to move the creation and url_reachability to main (maybe in a function) and then call profile_handler by passing the telemetry_data_collector as an argument.
Let me know if that seems like a good way to go?

Reply:
I like that approach more.

Reply:
I moved this part to main.py and renamed and modified the corresponding test.